File: CS/DXSample.Module/Master.cs
using System;

using DevExpress.Xpo;

using DevExpress.ExpressApp;
using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;
using DevExpress.Persistent.Validation;
using DevExpress.Data.Filtering;
using System.ComponentModel;

namespace DXSample.Module {
    [DefaultClassOptions]
    [DefaultProperty("MasterName")]
    public class Master : BaseObject {
        public Master(Session session) : base(session) { }
        private string _MasterName;
        public string MasterName {
            get { return _MasterName; }
            set { SetPropertyValue("MasterName", ref _MasterName, value); }
        }
        [Association("Master-Details")]
        public XPCollection<Detail> Details {
            get {
                return GetCollection<Detail>("Details");
            }
        }
    }

}

File: VB/DXSample.Module/Master.vb
Imports DevExpress.Xpo
Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.BaseImpl
Imports System.ComponentModel

Namespace DXSample.Module

    <DefaultClassOptions>
    <DefaultProperty("MasterName")>
    Public Class Master
        Inherits BaseObject

        Public Sub New(ByVal session As Session)
            MyBase.New(session)
        End Sub

        Private _MasterName As String

        Public Property MasterName As String
            Get
                Return _MasterName
            End Get

            Set(ByVal value As String)
                SetPropertyValue("MasterName", _MasterName, value)
            End Set
        End Property

        <Association("Master-Details")>
        Public ReadOnly Property Details As XPCollection(Of Detail)
            Get
                Return GetCollection(Of Detail)("Details")
            End Get
        End Property
    End Class
End Namespace
